.github/workflows/build.yml

```diff
@@ -0,0 +1,50 @@
+# Github action to build sphinx site and commit to gh-pages branch
+#
+#  Built site is pushed to 'gh-pages' branch
+#
+
+name: Sphinx Build
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  schedule:
+    - cron: "0 3 * * *" 
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+        
+    - name: Checkout page source
+      uses: actions/checkout@v3
+      
+    - name: Setup Python
+      uses: actions/setup-python@v3
+
+    - name: Install sphinx
+      run: pip3 install --user -r requirements.txt
+      
+    - name: delete _data folder
+      run: rm -Rf _data
+      
+    - name: download _data
+      run: mkdir _data && cd _data && gh run download -n _data
+      env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      
+    - name: Build page
+      run: make html
```

Comment:
Should we use *dirhtml* instead? This creates `/en/learn/index.html` instead of `/en/learn/index.html` and so on.

The output will be in `build/dirhtml` instead of `build/html`.

Reply:
sir, to use **_dirhtml_** we would have to define dirhtml command in **_Makefile_**  and we would also have to change the directory structures for various sections like learn and also the navigation of pages  while the pages for the site like https://henilp105.github.io/webpage/en/index ( without '.html' extension ) can be used without the need of dirhtml. but it can surely be implemented.

Thanks and Regards,
Henil Shalin Panchal